omegatrade-app/frontend/src/app/register/register.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ValidationService } from '../shared/validation.service';
+import { RestService } from '../auth/rest.service';
+import { SocialAuthService } from 'angularx-social-login';
+import { GoogleLoginProvider } from 'angularx-social-login';
+import { TokenStorageService } from '../auth/token-storage.service';
+import { SnackBarService } from '../shared/snackbar.service';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.css']
+})
+export class RegisterComponent implements OnInit {
+  signUpForm: any;
+  loader = false;
+  user: any;
+  // tslint:disable-next-line: max-line-length
+  constructor(private snackBarService: SnackBarService, private tokenStorage: TokenStorageService, private formBuilder: FormBuilder, private restService: RestService, private router: Router, private authService: SocialAuthService) {
+    // Init sign-up form with form builder.
+    this.signUpForm = this.formBuilder.group({
+      fullName: ['', [Validators.required]],
+      businessEmail: ['', [Validators.required, ValidationService.emailValidator]],
+      password: ['', [Validators.required]],
+      confirmPassword: ['', [Validators.required]],
+      provider: [''],
+      photoUrl: ['']
+    });
+  }
+
+  /**
+   *  Function to Initiate component.
+   */
+  ngOnInit(): void {
+    // Function to Subscribe to the authentication state.
+    // Receive a SocialUser object when the user logs in and a null when the user logs out.
+    this.authService.authState.subscribe((user) => {
```

Comment:
`this.authService.authState.subscribe((user) => {`
Above line will create a subscriber on this.authService.authState, which continuous look for any change in that variable, we should unsubscribe it inside ngOnDestroy method of this component to avoid unnecessary computations

Reply:
This function is removed as the auth User is handled in backend.

Reply:
This function is removed as the auth User is handled in backend.